drivers/pwm/pwm_capture.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020 Vestas Wind Systems A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <zephyr.h>
+#include <drivers/pwm.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(pwm_capture, CONFIG_PWM_LOG_LEVEL);
+
+struct z_pwm_pin_capture_cb_data {
+	uint32_t period;
+	uint32_t pulse;
+	struct k_sem sem;
+	int status;
+};
+
+static void z_pwm_pin_capture_cycles_callback(const struct device *dev,
+					      uint32_t pwm,
+					      uint32_t period_cycles,
+					      uint32_t pulse_cycles,
+					      int status,
+					      void *user_data)
+{
+	struct z_pwm_pin_capture_cb_data *data = user_data;
+
+	data->period = period_cycles;
+	data->pulse = pulse_cycles;
+	data->status = status;
+
+	k_sem_give(&data->sem);
+}
+
+int z_impl_pwm_pin_capture_cycles(const struct device *dev, uint32_t pwm,
+				  pwm_flags_t flags, uint32_t *period,
+				  uint32_t *pulse, k_timeout_t timeout)
+{
+	struct z_pwm_pin_capture_cb_data data;
+	int err;
+
+	if (flags & PWM_CAPTURE_MODE_CONTINUOUS) {
```

Comment:
This idiom is fragile: it works only because continuous is the non-zero value for the bitfield.  In that case the subsequent bitwise or of single has no effect.

I'd rather see these sorts of things as `(flags & PWM_CAPTURE_MODE_MASK) == PWM_CAPTURE_MODE_CONTINUOUS)` as that's very clear.

Reply:
Thanks, fixed.